lib/deaf_grandma.rb

```diff
@@ -19,9 +19,17 @@ def run!
 
 
   def speak(input)
-
-    #Implement your code here <<<<<<<<<
-
+    upper = /[A-Z]+/
+    lower = /[A-Z]?[a-z]+/
+    bye = /\A(BYE ?){3}\z/
```

Comment:
These three regular expressions could be extracted to constants, since they are the same every time this method is called.